backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/XmlFunctions.java

```diff
@@ -0,0 +1,91 @@
+package de.bund.digitalservice.ris.norms.application.service;
+
+import java.io.StringReader;
+import java.util.Optional;
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.InputSource;
+
+/** Namespace for functions dealing with XML Documents or Nodes */
+public class XmlFunctions {
+
+  // prevent default constructor
+  private XmlFunctions() {}
+
+  /**
+   * Create an XML Document from an xml string representation.
+   *
+   * @param xmlText an XML represented as a string
+   * @return a Document representation of the string's XML
+   */
+  public static Optional<Document> stringToXmlDocument(String xmlText) {
+    try {
+      final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+
+      // XXE vulnerability hardening, cf. https://www.sonarsource.com/blog/secure-xml-processor/
+      factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+      factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+      factory.setExpandEntityReferences(false);
+
+      final DocumentBuilder builder = factory.newDocumentBuilder();
+      final InputSource is = new InputSource(new StringReader(xmlText));
+      final Document document = builder.parse(is);
+      return Optional.of(document);
+    } catch (Exception e) {
```

Comment:
Why don't you just throw the exception in the method? This functionality will be called through a controller, for now, and we have a spring @ControllerAdvice catching all exceptions being thrown in all use cases. Also  catching the general "Exception" is not recommended as best practice, since it may also catch unexpected runtime exceptions including those you didn't anticipate.
Also catching an exception and not even logging anything would make it more difficult to investigate in case of errors